lib/stat_tracker.rb

```diff
@@ -87,4 +85,82 @@ def average_goals_by_season
     avg_goals
   end
 
+  def goals_for_average(team_id, filter = nil)
+    goals, games = 0, 0
+    if !filter
+      @game_teams_data.each do |game|
+        goals += game.goals if game.team_id == team_id
+        games += 1 if game.team_id == team_id
+      end
+    else
+      @game_teams_data.each do |game|
+        goals += game.goals if game.team_id == team_id && game.hoa == filter
+        games += 1 if game.team_id == team_id && game.hoa == filter
+      end
+    end
+    (goals.to_f / games).round(2)
+  end
+
+  def goals_against_average(team_id)
+    goals, games = 0, 0
+    @games_data.each do |game|
+      if game.home_team_id == team_id
+        goals += game.away_goals
+        games += 1
+      elsif game.away_team_id == team_id
+        goals += game.home_goals
+        games += 1
+      end
+    end
+    (goals.to_f / games).round(2)
+  end
```

Comment:
I think the two methods above (avg goals for & against) are probably strong refactor candidates given their length... but I'm not immediately certain how to do so in a still-readable/functional format.

Also on these two - I didn't call on the average method from calculable module because it doesn't specifically save any lines of code. I'm open to doing so if we think there's a valid reason for it.

Reply:
I think calling `average` is nicer to read as it makes it very clear what's happening on that line. It also keeps us consistent between methods.